generators/client/templates/src/main/webapp/app/blocks/config/_compile.config.js

```diff
@@ -5,13 +5,15 @@
         .module('<%=angularAppName%>')
         .config(compileServiceConfig);
 
-    compileServiceConfig.$inject = ['$compileProvider', 'ENV'];
+    compileServiceConfig.$inject = ['$compileProvider'];
 
-    function compileServiceConfig($compileProvider, ENV) {
+    function compileServiceConfig($compileProvider) {
         // disable debug data on prod profile to improve performance
-        if(ENV === 'prod'){
-            $compileProvider.debugInfoEnabled(false);
-        }
+        
+        // TODO: Howto fetch the dynamic profile in this config.... can this be done elsewhere ?
```

Comment:
Thats a hacky solution and will attract angular nazi :( @mraible any idea
on how to do this?
On 28 Apr 2016 00:46, "Davy De Waele" <notifications@github.com> wrote:

In
generators/client/templates/src/main/webapp/app/blocks/config/_compile.config.js
<https://github.com/jhipster/generator-jhipster/pull/3474#discussion_r61293501>
:

>          // disable debug data on prod profile to improve performance
> -        if(ENV === 'prod'){
> -            $compileProvider.debugInfoEnabled(false);
> -        }
> +
> +        // TODO: Howto fetch the dynamic profile in this config.... can this be done elsewhere ?

@deepu105 <https://github.com/deepu105> : The compileProvider can only be
accessed in config block.

You can pull the compileProvider out of the config block and use it in the
run method.

(function() {
    'use strict';

    var provider = {};

    angular
        .module('ribbon8App', [
            'ngStorage',
            'ngResource',
            'ngCookies',
            'ngAria',
            'ngCacheBuster',
            'ngFileUpload',
            'ui.bootstrap',
            'ui.bootstrap.datetimepicker',
            'ui.router',
            'infinite-scroll',
            // jhipster-needle-angularjs-add-module JHipster will add
new module here
            'angular-loading-bar'
        ])
        .config(config)
        .run(run);

    config.$inject = ['$compileProvider'];

    function config($compileProvider) {
        angular.copy($compileProvider, provider);
    }

    run.$inject = ['stateHandler','ActiveProfiles'];

    function run(stateHandler,ActiveProfiles) {
        stateHandler.initialize();

        ActiveProfiles.activeProfiles().then(function(response) {
            provider.debugInfoEnabled(!response.inProduction);
        });
    }
})();

—
You are receiving this because you were mentioned.

Reply to this email directly or view it on GitHub
<https://github.com/jhipster/generator-jhipster/pull/3474/files/bfb301cbc2770722e89419177c2e373c21d1c1ab#r61293501>